youtube_dl/extractor/animategamer.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import random
+
+from .common import InfoExtractor
+from ..utils import ExtractorError
+
+
+class AnimateGamerIE(InfoExtractor):
+    _VALID_URL = r'https://ani\.gamer\.com\.tw/animeVideo.php\?sn=(?P<id>[0-9]+)'
+    _TEST = {
+        'url': 'https://ani.gamer.com.tw/animeVideo.php?sn=11441',
+        'md5': '00d08c66bf9a998f8b13e4882277a002',
+        'info_dict': {
+            'id': '11441',
+            'ext': 'ts',
+            'title': '巴哈姆特之怒 -Manaria Friends-[1]',
+        },
+    }
+
+    def _extract_m3u8_formats_with_http_headers(self, m3u8_url, video_id, ext=None,
```

Comment:
This should be done in `_extract_m3u8_formats`.

Reply:
Thanks for reviewing the code.

I didn't find the way to make request with headers by `_extract_m3u8_formats()`.
Do you mean overridng it directly?

Reply:
You should add that ability since it's a generic need.